test/functional/sqlsrv/MsCommon.inc

```diff
@@ -269,9 +269,13 @@ function createUniqueIndexEx($conn, $tableName, $tableIndex, $colIndex)
     trace(" completed successfully.\n");
 }
 
-function dropTable($conn, $tableName)
+function dropTable($conn, $tableName, $check = false)
 {
-    $stmt = sqlsrv_query($conn, "IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'" . $tableName . "') AND type in (N'U')) DROP TABLE $tableName");
+    if ($check) {
+        $stmt = sqlsrv_query($conn, "IF EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'" . $tableName . "') AND type in (N'U')) DROP TABLE [$tableName]");
+    } else {
+        $stmt = sqlsrv_query($conn, "DROP TABLE [$tableName]");
+    }
```

Comment:
Do you ever use `$check`? Is it really necessary?

Reply:
yes I used it when creating a new table and in most other cases we don't check first, just simply drop the table :)

Reply:
I still think checking it first is better so the test won't fail if somehow there's a table of the same name in the database

Reply:
But you see the original method (back before we started refactoring) didn't even bother checking if this operation failed or not.

Reply:
I understand the original method only drops the table. But checking if the table exists doesn't take that much longer and it ensures us no error is thrown if the database isn't clean (e.g., execution of a test isn't complete and the table still remains).